### ISSUE TITLE
Exposing `textureCompress` for individual textures

### ISSUE THREAD

**Issue by marwie** [donmccurdy/glTF-Transform#853]
**Is your feature request related to a problem? Please describe.**
I'd like to use existing compression functions for individual textures/primitives/...

**Describe the solution you'd like**
Similarly to https://github.com/donmccurdy/glTF-Transform/pull/767 It would be nice to be able to run the `textureCompress` function on individual textures (providing the texture compression options per texture?)

**Describe alternatives you've considered**
Copying and modifying the code I need (e.g. the inner textures.map loop of `textureCompress` to apply webp compression to individual textures

**Additional context**
https://github.com/donmccurdy/glTF-Transform/blob/32dc66e8d5e2019f52bdcc6dbb0043d837007708/packages/functions/src/texture-compress.ts#L107

**Comment by donmccurdy** [comment]
Related:

- #845
- #801

I think we could do a `textureCompress` function, yes — would you be up for writing a PR?

It's worth noting that texture compression can be done in less code than I'm using, and `@gltf-transform/functions` will never expose the entirety of Sharp's API. So being able to do skip the wrapping function and use Sharp directly is something that I hope users will feel very free to do, too:


```javascript
import sharp from 'sharp';
import { BufferUtils, HTTPUtils } from '@gltf-transform/core';

for (const texture of document.getRoot().listTextures()) {
  const buffer = await sharp(texture.getImage()).toFormat('webp', {lossless: true}).toBuffer();
  texture
    .setMimeType('image/webp')
    .setImage(BufferUtils.toView(buffer))
    .setURI(HTTPUtils.basename(texture.getURI()) + '.webp');
}

document.createExtension(EXTTextureWebP).setRequired(true);
```

**Comment by marwie** [comment]
Yes happy to PR this, just wanted to run some tests before to make sure I didnt break anything and it's working as intended.

Thanks for the code snippet. That's goo to know and maybe something I'll be doing then instead in our code and just handle the settings we ship in our extension (that being said having a fallback to the default gltf-transform compression function is still handy in cases where no explicit compression settings were assigned to a texture but we still want to apply defaults)

**Comment by donmccurdy** [comment]
Fixed by #859.

**Comment by leefsmp** [comment]
@donmccurdy could you provide a bit more details on how to handle the `webp` extension, it's not clear to me how I'm supposed to use the `EXTTextureWebP` ... thanks.
